java-frontend/src/test/files/se/DivisionByZeroCheck.java

```diff
@@ -495,3 +498,67 @@ private int g(int x, int y) {
     }
   }
 }
+
+class BigIntegerAndDecimal {
+  void simpleBigInt(BigInteger r) {
+    BigInteger z1 = BigInteger.valueOf(0); // flow@simpleBigInt {{'valueOf()' can return zero.}} flow@simpleBigInt {{Implies 'z1' can be zero.}}
+    BigInteger z2 = z1; // flow@simpleBigInt {{Implies 'z2' has the same value as 'z1'.}}
+    r = r.divide(z2); // Noncompliant [[flows=simpleBigInt]] {{Make sure "z2" can't be zero before doing this division.}} flow@simpleBigInt {{Division by zero.}}
+  }
+
+  void simpleBigDec(BigDecimal r) {
+    BigDecimal z1 = BigDecimal.valueOf(0);;
+    BigDecimal z2 = z1;
+    r = r.divide(z2); // Noncompliant
+  }
+
+  void bigMultiply(BigInteger r) {
+    BigInteger z1 = BigInteger.valueOf(0);
+    BigInteger b1 = BigInteger.valueOf(4);
+    BigInteger mult1 = z1.multiply(b1);
+
+    r = r.divide(mult1); // Noncompliant
+  }
+
+  void bigMultiply2(BigInteger r) {
+    BigInteger z1 = BigInteger.valueOf(0);
+    BigInteger b1 = BigInteger.valueOf(4);
+    BigInteger mult2 = b1.multiply(z1);
+
+    r = r.divide(mult2); // Noncompliant
+  }
+
+  void bigMultiplyChained(BigInteger r) {
+    BigInteger z1 = BigInteger.valueOf(0);
+    BigInteger b1 = BigInteger.valueOf(4);
+    r = r.divide(z1.multiply(b1)); // Noncompliant
+  }
+
+  void bigPlus(BigInteger r) {
+    BigInteger z1 = BigInteger.valueOf(0);
+    BigInteger z2 = BigInteger.valueOf(0);
+    BigInteger sum1 = z1.add(z2);
+
+    r = r.divide(sum1); // Noncompliant
+  }
+
+  void bigMinus(BigInteger r) {
+    BigInteger z1 = BigInteger.valueOf(0);
+    BigInteger z2 = BigInteger.valueOf(0);
+    BigInteger sum1 = z1.subtract(z2);
+
+    r = r.divide(sum1); // Noncompliant
+  }
+
+  void bigRemainder(BigInteger r) {
+    BigInteger z1 = BigInteger.valueOf(0);
+    BigInteger v1 = BigInteger.valueOf(42);
+    BigInteger z2 = z1.multiply(v1);
+    r = r.remainder(z2); // Noncompliant
+  }
+
+  void simpleBigInt(BigInteger r, int value) {
+    BigInteger z1 = BigInteger.valueOf(value);
+    r = r.divide(z1); // Compliant
+  }
+}
```

Comment:
I addition to the above use cases, we should probably also support:

- The BigInteger.ZERO and BigDecimal.ZERO identifiers
```java
  void zeroConstantBigInt(BigInteger r) {
    BigInteger z1 = BigInteger.ZERO;
    r = r.divide(z1); // Noncompliant
  }
```
- Keeping "ZERO" constraint through "valueOf" from long, double to BigInteger, BigDecimal
```java
  void valueOfBigInt(BigInteger r) {
    long z1 = 0;
    BigInteger z2 = BigInteger.valueOf(z1);
    r = r.divide(z2); // Noncompliant
  }
```
- Keeping "ZERO" constraint through BigDecimal constructor from BigInteger, double, int, long
```java
  void constructorBigDecimal(BigDecimal r) {
    double z1 = 0;
    BigDecimal z2 = new BigDecimal(z1);
    r = r.divide(z2); // Noncompliant
  }
```
 - Keeping "ZERO" constraint through "toBigInteger", "toBigIntegerExact", "shortValueExact", "round", "	remainder", "pow", "longValueExact", "longValue" , "	intValueExact", "intValue", "floatValue", "doubleValue", "byteValueExact", "abs", 	shiftLeft, shiftRight
 - Support 	BigDecimal.divideToIntegralValue like BigDecimal.divide

Reply:
Good point! I added one commit for each case that you stated.

Reply:
Nice job!
Unfortunately, now we raise a FP on the following code:
```java
    BigDecimal x = new BigDecimal(0);
    BigDecimal y = new BigDecimal(0);
    if (x == y) { // false-positive "always true"
      System.out.println("1");
    } else {
      System.out.println("2"); // <==
    }
```